qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -411,9 +416,19 @@ def _run(self) -> 'VQEResult':
         self._quantum_instance.circuit_summary = True
 
         self._eval_count = 0
+
+        # Convert the gradient operator into a callable function that is compatible with the
+        # optimization routine.
+        if self._gradient:
```

Comment:
I think this block should be inside `VQAlgorithm` or not? Otherwise QAOA (or the base class `VQAlgorithm`) doesn't cover the case of `gradient` being a `GradientBase`.

Reply:
The problem is that the Gradient requires the Ansatz as well as the Hamiltonian/observable to construct the gradient object. However, the Hamiltonian/observable is not known on the level of the VQAlgorithm.

Reply:
Ok but then `GradientBase` is not supported in the `VQAlgorithm`s init, right? Since in QAOA the operator is available we should it add there though, or not?

Reply:
Exactly. 
That should be handled since VQE acts as base class for QAOA.

Reply:
oh of course! my bad, thanks!